python/instrumentation/openinference-instrumentation-crewai/src/openinference/instrumentation/crewai/_wrappers.py

```diff
@@ -0,0 +1,256 @@
+    output regardless of whether the those inputs are passed as positional or
+    keyword arguments.
+    """
+
+    # For typical class methods, the corresponding instance of inspect.Signature
+    # does not include the self parameter. However, the inspect.Signature
+    # instance for __call__ does include the self parameter.
+    method_signature = signature(method)
+    first_parameter_name = next(iter(method_signature.parameters), None)
+    signature_contains_self_parameter = first_parameter_name in ["self"]
+    bound_arguments = method_signature.bind(
+        *(
+            [None]  # the value bound to the method's self argument is discarded below, so pass None
+            if signature_contains_self_parameter
+            else []  # no self parameter, so no need to pass a value
+        ),
+        *args,
+        **kwargs,
+    )
+    return safe_json_dumps(
+        {
+            **{
+                argument_name: argument_value
+                for argument_name, argument_value in bound_arguments.arguments.items()
+                if argument_name not in ["self", "kwargs"]
+            },
+            **bound_arguments.arguments.get("kwargs", {}),
+        },
+        cls=SafeJSONEncoder,
+    )
+
+
+class _ExecuteCoreWrapper:
+    def __init__(self, tracer: trace_api.Tracer) -> None:
+        self._tracer = tracer
+
+    def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        if instance:
+            span_name = f"{instance.__class__.__name__}.{wrapped.__name__}"
+        else:
+            span_name = wrapped.__name__
+        with self._tracer.start_as_current_span(
+            span_name,
+            attributes=dict(
+                _flatten(
+                    {
+                        OPENINFERENCE_SPAN_KIND: OpenInferenceSpanKindValues.AGENT,
+                        SpanAttributes.INPUT_VALUE: _get_input_value(
+                            wrapped,
+                            *args,
+                            **kwargs,
+                        ),
+                    }
+                )
+            ),
+        ) as span:
+            try:
+                agent = args[0]  # Assuming the first argument is the instance
+                crew = agent.crew if agent else None
+                task = instance
+
+                if crew:
+                    span.set_attribute("crew_key", crew.key)
+                    span.set_attribute("crew_id", str(crew.id))
+                span.set_attribute("task_key", task.key)
+                span.set_attribute("task_id", str(task.id))
+
+                if crew and crew.share_crew:
+                    span.set_attribute("formatted_description", task.description)
+                    span.set_attribute("formatted_expected_output", task.expected_output)
+
+                response = wrapped(*args, **kwargs)
```

Comment:
could the `try` be here instead? fewer indents that way
```suggestion
            try:
                response = wrapped(*args, **kwargs)
```

Reply:
done!